rlclientlib/dedup.cc

```diff
@@ -253,36 +277,59 @@ struct dedup_collection_serializer
 class dedup_extensions : public logger::i_logger_extensions
 {
 public:
-	dedup_extensions(const utility::configuration& c, i_time_provider* time_provider) : logger::i_logger_extensions(c), _dedup_state(c, time_provider) {}
+	dedup_extensions(const utility::configuration& c, bool use_compression, bool use_dedup, i_time_provider* time_provider) :
+    logger::i_logger_extensions(c), _dedup_state(c, use_compression, use_dedup, time_provider), _use_dedup(use_dedup), _use_compression(use_compression) {}
 
 	logger::i_async_batcher<generic_event>* create_batcher(logger::i_message_sender* sender, utility::watchdog& watchdog,
 																									error_callback_fn* perror_cb, const char* section) override {
 		auto config = utility::get_batcher_config(_config, section);
 		int _dummy = 0;
-		return new logger::async_batcher<generic_event, dedup_collection_serializer>(
-				sender,
-				watchdog,
-				_dedup_state,
-				perror_cb,
-				config);
+    if(_use_dedup) {
```

Comment:
`if(_use_dedup || _use_compression)`?

Reply:
This controls whether dedup_collection_serializer is used, which is a decorator on top of fb_collection_serializer that injects the dedup dictionary into the batch.